src/core/Akka.Streams/Implementation/Fusing/Ops.cs

```diff
@@ -2981,17 +2983,7 @@ public override void PreStart()
                     _log = _stage._adapter;
                 else
                 {
-                    try
-                    {
-                        var materializer = ActorMaterializerHelper.Downcast(Materializer);
-                        _log = new BusLogging(materializer.System.EventStream, _stage._name, GetType(), materializer.System.Settings.LogFormatter);
-                    }
-                    catch (Exception ex)
-                    {
-                        throw new Exception(
-                            "Log stage can only provide LoggingAdapter when used with ActorMaterializer! Provide a LoggingAdapter explicitly or use the actor based flow materializer.",
-                            ex);
-                    }
+                    _log = Log;
```

Comment:
Simplify the built-in `.Log` stage to just use the built-in `Log` rather than doing all of this weird stuff.

Reply:
@Danthar it comes from the `GraphStage.Log` property, which is the abstract base class for all stream shapes. It's lazily instantiated and gets created by the `ActorMaterializer` only when it's called.

Reply:
It's still using `BusLogging` et al - it's just doing a more manual version of `Context.GetLogger()` because there's no guarantee that the the `StageRef` for this `GraphStage` is available (for reasons I don't quite understand, but that came up during testing.)

Reply:
Ah yep found it.
That call eventually ends up at:
       `     return new BusLogging(context.System.EventStream, logSource.Source, logSource.Type, logMessageFormatter ?? context.System.Settings.LogFormatter);`
            
            So to answer my own question. Yes it does seems to do the same.